src/client-side-encryption/state_machine.ts

```diff
@@ -543,15 +586,21 @@ export class StateMachine {
   fetchKeys(
     client: MongoClient,
     keyVaultNamespace: string,
-    filter: Uint8Array
+    filter: Uint8Array,
+    timeoutContext?: TimeoutContext
   ): Promise<Array<DataKey>> {
     const { db: dbName, collection: collectionName } =
       MongoDBCollectionNamespace.fromString(keyVaultNamespace);
 
+    if (timeoutContext?.csotEnabled() && timeoutContext?.remainingTimeMS <= 0) {
+      throw new MongoOperationTimeoutError('Timed out before dataKey fetched.');
+    }
     return client
       .db(dbName)
       .collection<DataKey>(collectionName, { readConcern: { level: 'majority' } })
-      .find(deserialize(filter))
+      .find(deserialize(filter), {
+        timeoutMS: timeoutContext?.csotEnabled() ? timeoutContext?.remainingTimeMS : undefined
```

Comment:
The default timeout mode is LIFETIME - but just to be on the safe side, can we pass in `timeoutMode: CursorTimeoutMode.LIFETIME` too? That prevents the timeout behavior in these functions from silently changing if the default timeout mode ever changes.

Reply:
Sounds good!